webportal/webapp/src/factories/window.factory.ts

```diff
@@ -0,0 +1,4 @@
+import { InjectionToken } from '@angular/core';
```

Comment:
Stackoverflow told me this is the best way to inject `window`... In tests now, `window` can be mocked out.